app/app.rb

```diff
@@ -0,0 +1,73 @@
+require 'sinatra/base'
+require 'sinatra/flash'
+require_relative './models/user'
+require_relative './models/peep'
+require_relative 'data_mapper_setup'
+
+class ChitterChatter < Sinatra::Base
+  use Rack::MethodOverride
+  register Sinatra::Flash
+  enable :sessions
+  set :session_secret, 'super secret'
+
+  get '/' do
+    @peeps = Peep.all.reverse!
+    erb :index
+  end
+
+  get '/sign-up' do
+    @user = User.new
+    erb :sign_up
+  end
+
+  helpers do
+    def current_user
+      @current_user ||= User.get(session[:user_id])
+    end
+  end
+
+  post '/users' do
+    @user = User.create(email: params[:email],
+    password: params[:password],
+    password_confirmation: params[:password_confirmation],
+    name: params[:name],
+    username: params[:username])
+    if @user.save
+      session[:user_id] = @user.id
```

Comment:
Could the lines than set session user_id be refactored to use a helper function eg:

```
def log_in_user(user_id)
  session["user_id"] = user_id
end
```

That way, the idea of how to log in a user is abstracted into one place.  This makes the mechanism of logging in a user easier to change.  It also makes it clearer what the code is doing.